File: vinilos_g24/app/src/main/java/RecyclerViewAdaptador.java
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class RecyclerViewAdaptador extends RecyclerView.Adapter<RecyclerViewAdaptador.ViewHolder> {

    public static class  ViewHolder extends RecyclerView.ViewHolder{
        private TextView nombreAlbum,cantante;
        ImageView imgAlbum;


        public ViewHolder(@NonNull View itemView) {
            super(intemView);
            nombreAlbum = (TextView)itemView.findViewById(R.id.tvAlbum);
            cantante = (TextView)itemView.findViewById(R.id.tvCantante);
            imgAlbum = (ImageView)itemView.findViewById(R.id.imageAlbum);
        }
    }

    public List<CantanteModelo> albumLista;

    public RecyclerViewAdaptador(List<CantanteModelo> albumLista) {
        this.albumLista = albumLista;
    }


    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_cover,parent,false);
        ViewHolder viewHolder = new ViewHolder(view);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.nombreAlbum.setText(albumLista.get(position).getNombreAlbum());
        holder.cantante.setText(albumLista.get(position).getCantante());
        holder.imgAlbum.setImageResource(albumLista.get(position).getImgAlbum());
    }

    @Override
    public int getItemCount() {
        return albumLista.size();
    }
}

File: vinilos_g24/app/src/main/java/RecyclerViewAdaptador.kt
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class RecyclerViewAdaptador(private val albumLista: List<CantanteModelo>) :
    RecyclerView.Adapter<RecyclerViewAdaptador.ViewHolder>() {

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val nombreAlbum: TextView = itemView.findViewById(R.id.tvAlbum)
        val cantante: TextView = itemView.findViewById(R.id.tvCantante)
        val imgAlbum: ImageView = itemView.findViewById(R.id.imageAlbum)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_cover, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.nombreAlbum.text = albumLista[position].nombreAlbum
        holder.cantante.text = albumLista[position].cantante
        holder.imgAlbum.setImageResource(albumLista[position].imgAlbum)
    }

    override fun getItemCount(): Int {
        return albumLista.size
    }
}